odoo/addons/base/models/res_company.py

```diff
@@ -182,6 +183,46 @@ def _onchange_parent_id(self):
                 if self[fname] != self.parent_id[fname]:
                     self[fname] = self.parent_id[fname]
 
+    @api.depends('country_id')
+    def _compute_uninstalled_l10n_module_ids(self):
+        # This will only compute uninstalled modules with auto-install without recursion,
+        # the rest will eventually be handled by `button_install`
+        self.env['ir.module.module'].flush_model(['auto_install', 'country_ids', 'dependencies_id'])
+        self.env['ir.module.module.dependency'].flush_model()
+        self.env.cr.execute("""
+            SELECT country.id,
+                   ARRAY_AGG(module.id)
+              FROM ir_module_module module,
+                   res_country country
+             WHERE module.auto_install
+               AND state NOT IN %(install_states)s
+               AND NOT EXISTS (
+                       SELECT 1
+                         FROM ir_module_module_dependency d
+                         JOIN ir_module_module mdep ON (d.name = mdep.name)
+                        WHERE d.module_id = module.id
+                          AND d.auto_install_required
+                          AND mdep.state NOT IN %(install_states)s
+                   )
+               AND EXISTS (
+                       SELECT 1
+                         FROM module_country mc
+                        WHERE mc.module_id = module.id
+                          AND mc.country_id = country.id
+                   )
```

Comment:
If someone could build a system where you can link tables.

Reply:
I would suggest that you go and comment on these commits as well.
https://github.com/odoo/odoo/commit/fce64e7d94a40333a6c2eb369ee5a94f9bd987c3
https://github.com/odoo/odoo/commit/256eae55b494238dc9481ac404d0a9d6e5f22465

Because clearly these people deserve this kind of useful comments.

Reply:
EXISTS VS NOT EXISTS, not the same .

Here we have something like
`SELECT ... FROM A,B .... WHERE EXISTS (SELECT 1 FROM C WHERE C.A_id=A.id AND C.B_id=B.id`

which can be
` SELECT ... FROM A JOIN C ON C.A_id=A.id  JOIN B ON C.B_id=B.id`
But the second form is optimized.


The NOT EXISTS is equivalent to 
` SELECT FROM A LEFT JOIN B on ... WHERE B.id IS NULL` 
Both are executed the same way in PostgreSQL AFAIK.

Reply:
> AFAIK

Yes, AFAYK.
The `EXISTS` and `NOT EXISTS` have different optimizations compared to joining. For instance they don't run if other (less expensive) logical branches can validate or invalidate the condition.
Don't make assumptions without testing it.
In any case, I don't believe this query deserves intensive performance analysis. It is also more readable since we are using a `EXISTS` and a `NOT EXISTS` in the same query for related stuff.

If you believe something deserves to be improved, please make a PR to fix it.